### ISSUE TITLE
Unexpected result involving `scan` and `vmap`

### ISSUE THREAD

**Issue by puct9** [jax-ml/jax#36958]
### Description

The issue was introduced between 0.9.2 and 0.10.0 because the former version does not exhibit the unexpected behaviour.

This _relatively simple snippet_ does not produce the expected output, and the result is instead **egregiously inaccurate**:

```py
import jax

def forward(weight, bias, x):
    return weight @ x + bias

def rollout_bug(weight, bias, key, points: int, steps: int):
    dt = 1 / steps

    def rollout_step(x, _):
        vf = jax.vmap(forward, (None, None, 0))(weight, bias, x)
        return x + dt * vf, x

    res, _ = jax.lax.scan(
        rollout_step,
        jax.random.normal(key, (points, 2)),
        length=steps,
    )
    return res
```

If we return the variable `vf` within the `rollout_step` function, then we are OK.

```py
def rollout_ok(weight, bias, key, points: int, steps: int):
    dt = 1 / steps

    def rollout_step(x, _):
        vf = jax.vmap(forward, (None, None, 0))(weight, bias, x)
        return x + dt * vf, (x, vf)  # the difference is that we return `vf` variable for fun

    res, _ = jax.lax.scan(
        rollout_step,
        jax.random.normal(key, (points, 2)),
        length=steps,
    )
    return res
```

We can confirm our results by comparing against a basline manual implementation.

```py
def rollout_manual(weight, bias, key, points: int, steps: int):
    dt = 1 / steps
    x = jax.random.normal(key, (points, 2))
    for _ in range(steps):
        x = x + dt * jax.vmap(forward, (None, None, 0))(weight, bias, x)
    return x
```

You can inspect the incorrect behaviour by running the code like so:

```py
import matplotlib.pyplot as plt

wkey, bkey = jax.random.split(jax.random.PRNGKey(0))
lim = 2**-0.5
weight = jax.random.uniform(wkey, (2, 2), minval=-lim, maxval=lim)
bias = jax.random.uniform(bkey, (2,), minval=-lim, maxval=lim)

res_bug = rollout_bug(weight, bias, jax.random.PRNGKey(0), 64, 10)
plt.scatter(*res_bug.T)
plt.show()

res_ok = rollout_ok(weight, bias, jax.random.PRNGKey(0), 64, 10)
plt.scatter(*res_ok.T)
plt.show()

res_manual = rollout_manual(weight, bias, jax.random.PRNGKey(0), 64, 10)
plt.scatter(*res_manual.T)
plt.show()
```

While the plots look visually similar, the points are actually quite far from each other, evidenced by the xy axis ticks. There are also differences in the relative positions of the points.

Other interesting discoveries:

- Issue not present on CPU
- Issue not present when `points < 48` 
- Issue not present without the use of `vmap`

Colab notebook demo: https://colab.research.google.com/drive/1wJ3cVbcrOl5wfutqJ2YF4LVMjeRZLAlV?usp=sharing

### System info (python version, jaxlib version, accelerator, etc.)

Reproduced on local WSL RTX 2060 Super and cloud Google Colab T4.

Local:
```
E0418 20:39:07.278345   28199 cuda_executor.cc:1526] Could not get kernel mode driver version: (INVALID_ARGUMENT: Version does not match the format X.Y.Z)
E0418 20:39:07.286539   28118 cuda_executor.cc:1526] Could not get kernel mode driver version: (INVALID_ARGUMENT: Version does not match the format X.Y.Z)
jax:    0.10.0
jaxlib: 0.10.0
numpy:  2.4.4
python: 3.13.5 (main, Jul 11 2025, 22:43:46) [Clang 20.1.4 ]
device info: NVIDIA GeForce RTX 2060 SUPER-1, 1 local devices"
process_count: 1
platform: uname_result(system='Linux', node='DESKTOP-O2TEBKS', release='6.6.87.2-microsoft-standard-WSL2', version='#1 SMP PREEMPT_DYNAMIC Thu Jun  5 18:30:46 UTC 2025', machine='x86_64')

$ nvidia-smi
Sat Apr 18 20:39:07 2026       
+-----------------------------------------------------------------------------------------+
| NVIDIA-SMI 570.124.04             Driver Version: 572.61         CUDA Version: 12.8     |
|-----------------------------------------+------------------------+----------------------+
| GPU  Name                 Persistence-M | Bus-Id          Disp.A | Volatile Uncorr. ECC |
| Fan  Temp   Perf          Pwr:Usage/Cap |           Memory-Usage | GPU-Util  Compute M. |
|                                         |                        |               MIG M. |
|=========================================+========================+======================|
|   0  NVIDIA GeForce RTX 2060 ...    On  |   00000000:29:00.0  On |                  N/A |
|  0%   45C    P2             43W /  175W |    7529MiB /   8192MiB |     23%      Default |
|                                         |                        |                  N/A |
+-----------------------------------------+------------------------+----------------------+
                                                                                         
+-----------------------------------------------------------------------------------------+
| Processes:                                                                              |
|  GPU   GI   CI              PID   Type   Process name                        GPU Memory |
|        ID   ID                                                               Usage      |
|=========================================================================================|
|  No running processes found                                                             |
+-----------------------------------------------------------------------------------------+
```

Colab T4:
```
jax:    0.10.0
jaxlib: 0.10.0
numpy:  2.0.2
python: 3.12.13 (main, Mar  4 2026, 09:23:07) [GCC 11.4.0]
device info: Tesla T4-1, 1 local devices"
process_count: 1
platform: uname_result(system='Linux', node='8ffca4395082', release='6.6.113+', version='#1 SMP Mon Feb  2 12:27:57 UTC 2026', machine='x86_64')

$ nvidia-smi
Sat Apr 18 10:41:18 2026       
+-----------------------------------------------------------------------------------------+
| NVIDIA-SMI 580.82.07              Driver Version: 580.82.07      CUDA Version: 13.0     |
+-----------------------------------------+------------------------+----------------------+
| GPU  Name                 Persistence-M | Bus-Id          Disp.A | Volatile Uncorr. ECC |
| Fan  Temp   Perf          Pwr:Usage/Cap |           Memory-Usage | GPU-Util  Compute M. |
|                                         |                        |               MIG M. |
|=========================================+========================+======================|
|   0  Tesla T4                       Off |   00000000:00:04.0 Off |                    0 |
| N/A   58C    P0             31W /   70W |   11313MiB /  15360MiB |      0%      Default |
|                                         |                        |                  N/A |
+-----------------------------------------+------------------------+----------------------+

+-----------------------------------------------------------------------------------------+
| Processes:                                                                              |
|  GPU   GI   CI              PID   Type   Process name                        GPU Memory |
|        ID   ID                                                               Usage      |
|=========================================================================================|
|    0   N/A  N/A            3033      C   /usr/bin/python3                      11310MiB |
+-----------------------------------------------------------------------------------------+
```

**Comment by puct9** [comment]
I'm unsure if I am missing something, but as far as I can tell, JAX is producing completely unexpected results in a rather tame program. I can work on reducing the reproduction example further, but in the meantime some colour would be much appreciated.

**Comment by puct9** [comment]
Colab notebook updated for JAX 0.10.1, will try to reduce the reproduction if I have capacity.

**Comment by mattjj** [comment]
Thanks for raising this, and the thorough reproduction. If the issue isn’t present on CPU it might be an xla:gpu miscompile. I’ll try to bisect it.

**Comment by puct9** [comment]
As of `0.11.1`, this issue seems to have disappeared.
